frontend/src/bundles/sign-up/components/first-step/first-step.tsx

```diff
@@ -0,0 +1,237 @@
+import {
+    useAppForm,
+    useCallback,
+    useEffect,
+    useState,
+} from '~/bundles/common/hooks/hooks.js';
+import { type UserSignUpStep1Dto } from '~/bundles/sign-up/types/types.js';
+import { signUpStep1ValidationSchema } from '~/bundles/sign-up/validation-schemas/validation-schemas.js';
+
+import { DEFAULT_SIGN_UP_PAYLOAD_STEP1 } from './constants/constants.js';
+
+type Properties = {
+    onSubmit: (payload: UserSignUpStep1Dto) => void;
+};
+const options = [
+    { value: 0, label: '0' },
+    { value: 20, label: '20' },
+    { value: 40, label: '40' },
+    { value: 60, label: '60' },
+    { value: 80, label: '80' },
+    { value: 100, label: '100' },
+];
+const optionsSelect = [
+    { value: 'Option', label: 'Option' },
+    { value: '20', label: '20' },
+    { value: '40', label: '40' },
+];
+const EMPTY_OBJECT_LENGTH = 0;
+const FirstStep: React.FC<Properties> = ({ onSubmit }) => {
+    const [hasError, setHasError] = useState<boolean>(false);
+    const { control, errors, handleSubmit } = useAppForm({
+        defaultValues: DEFAULT_SIGN_UP_PAYLOAD_STEP1,
+        validationSchema: signUpStep1ValidationSchema,
+    });
+
+    const handleFormSubmit = useCallback(
+        (formData: UserSignUpStep1Dto): void => {
+            onSubmit(formData);
+        },
+        [onSubmit],
+    );
+    useEffect(() => {
+        if (Object.keys(errors).length === EMPTY_OBJECT_LENGTH) {
+            setHasError(false);
+        } else {
+            setHasError(true);
+        }
+    }, [errors]);
+
+    const handleValidateBeforeSubmit = useCallback((): void => {
+        void handleSubmit(handleFormSubmit)();
+    }, [handleSubmit, handleFormSubmit]);
+
+    const handleCheckboxChange = useCallback(
+        (
+            field: ControllerRenderProps<UserSignUpStep1Dto, 'employmentTypes'>,
+            selectedValue: string,
+        ): void => {
+            const updatedValue = field.value.includes(selectedValue)
+                ? field.value.filter((item) => item !== selectedValue)
+                : [...field.value, selectedValue];
+            field.onChange(updatedValue);
+        },
+        [],
+    );
+
+    const handleCheckboxOnChange = useCallback(
+        (
+            field: ControllerRenderProps<UserSignUpStep1Dto, 'employmentTypes'>,
+            selectedValue: string,
+        ) =>
+            (): void => {
+                handleCheckboxChange(field, selectedValue);
+            },
+        [handleCheckboxChange],
+    );
+
+    const renderCheckboxes = useCallback(
+        ({
+            field,
+        }: {
+            field: ControllerRenderProps<UserSignUpStep1Dto, 'employmentTypes'>;
+            fieldState: ControllerFieldState;
+            formState: UseFormStateReturn<UserSignUpStep1Dto>;
+        }): React.ReactElement => {
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            const { ref, ...newField } = field;
```

Comment:
Why we need newField, can we pass full field object to Checkbox props ?

Reply:
@MaksymGavatuk  ref gives warning , and we don't need ref it so it is removed
Here is the link to conversation:
https://bsa2023.slack.com/archives/C05NY1E7CBS/p1693381909116219